pytorch_lightning/trainer/trainer.py

```diff
@@ -1259,3 +1261,10 @@ def _log_device_info(self) -> None:
                 "IPU available but not used. Set the `ipus` flag in your trainer"
                 " `Trainer(ipus=8)` or script `--ipus=8`."
             )
+
+    def _on_expection(self):
```

Comment:
I have a question about this function - is there a reason it is a function instead of a few lines in the exception handling? This function is only called once

This function seems a bit like a callback, but that's not what it is right? Still learning and understanding

Reply:
It was just meant as a stub for further improvements:

> We can think about the fault-tolerant checkpoint design in a later PR.

It's entirely fine to change it - although it wouldn't be a callback as it only needs to run on one hook: `on_exception`

Reply:
got it, thank you!!